src/compiler/gcc.rs

```diff
@@ -351,6 +352,8 @@ where
                 Argument::Raw(ref val) => {
                     if input_arg.is_some() {
                         multiple_input = true;
+                        multiple_input_files
+                            .push(val.to_str().iter().fold("".to_string(), |acc, s| acc + s))
```

Comment:
That's not what I meant, sorry for the lack of clarity. If the raw argument is not valid UTF8, I'd argue we should emit a not-cachable due to encoding (which I am not sure exists) result rather than trying to drop arguments.

Reply:
`val` is an `&OsString` and this needs to be a `String`, so it's all about conversion. The seemingly simplest way I found of doing that is dropping it to an `Option` and `unwrap`ping. Not sure if it is the best. The above seems to work as well, although it does seem a bit indirect.

Reply:
If `OsString::to_str()` returns `None`, that's an error. We do expect all of these to be UTF-8. So simply dropping arguments would modify the build process which is not ok, hence the suggestions to return non-cachable and compile it locally instead.

Reply:
I think we're not on the same page and hence the misunderstandings. It might very well be because of my very poor knowledge of Rust that I don't fully understand what you mean.

So lets take it step by step. I will try to explain what I was attempting to achieve here:
- `multiple_input_files` serves no other purpose than just being a vector for holding string arguments that were considered to be multiple input files by the current implementation
-  this vector is then flattened into a comma separated string upon generating the non-cachable error message
- no arguments that go through sccache are modified or dropped (unless my poor implementation does something to the contrary, which was not my intention - perhaps the removal of `val.copy()`?)
- in this particular piece of code we're discussing in this conversation I am attempting to add the parsed argument to the `multiple_input_files` vector; the `val` of the raw argument is an `OsString` type, while the vector expects `String`, hence the attempt to covert from the former to the latter; the initial implementation used `unwrap` because it was the most direct way I found to do the conversion after reading the Rust documentation on these types

I will be grateful for any suggestion on how to best achieve the type conversion described above as I believe this is the core issue here.

Reply:
I've reverted back to using `unwrap_or` this time around (to avoid the concern of a panic that can happen in `unwrap`). I really have no other idea how else to do this conversion in a sane way.